front/main/app/routes/infobox-builder.js

```
@@ -126,111 +191,68 @@ export default Ember.Route.extend(ConfirmationMixin, {
 	},
 
 	/**
-	 * @desc add oasis portable infobox styles to DOM
-	 * and add class to main application body to be
-	 * able to style whole containter as we want
+	 * Adds oasis portable infobox styles to the DOM and a class to the body element
 	 *
-	 * @param {Array} promiseResponseArray
+	 * @param {Object} serverResponse
 	 * @returns {Ember.RSVP.Promise}
 	 */
-	setupStyles(promiseResponseArray) {
+	setupStyles(serverResponse) {
 		Ember.$('body').addClass('infobox-builder-body-wrapper');
 
-		return new Ember.RSVP.Promise((resolve) => {
+		if (serverResponse.css) {
 			let html = '';
 
-			promiseResponseArray[0].css.forEach(
+			serverResponse.css.forEach(
 				(url) => {
 					html += `<link type="text/css" rel="stylesheet" href="${url}">`;
 				}
 			);
 
 			$(html).appendTo('head');
-
-			resolve(promiseResponseArray);
-		});
+		}
 	},
 
 	/**
-	 * @desc pass received from MW infobox state (for already existing infobox templates)
-	 * and set it as a property on controller. We don't need to keep this in model as it's
-	 * not connected with data we need there.
+	 * Accepts data received from MW and sets it on controller to be used in afterModel
 	 *
-	 * @param {Array} promiseResponseArray
-	 * @returns {Array} not modified promiseResponseArray
+	 * @param {Object} serverResponse
+	 * @returns {void}
 	 */
-	setupInfoboxState(promiseResponseArray) {
-		const assets = promiseResponseArray[0].data;
-		let templateData = {};
-
-		if (assets) {
-			templateData = JSON.parse(assets);
-			if (templateData.data) {
-				this.controllerFor('infobox-builder').set('templateData', templateData.data);
-			}
+	setupInfoboxData(serverResponse) {
+		const infoboxData = serverResponse.data,
+			controller = this.controllerFor('infobox-builder');
+		let infoboxDataParsed = {};
```

another thing is doing `JSON.parse` without `try catch`. Do we want to fail hard or have some graceful error fallback?